uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java

```diff
@@ -111,10 +112,18 @@ public void setGridPreferencesStore(final GridPreferencesStore gridPreferences)
     }
 
     public void saveGridPreferences() {
-        if (gridPreferencesStore != null && preferencesService != null) {
+        saveGridPreferences(null);
+    }
+
+    public void saveGridPreferences(Command command) {
+        if (gridPreferencesStore != null && preferencesService != null
+                && gridPreferencesStore.getGlobalPreferences().getKey() != null ) {
```

Comment:
@nmirasch I think avoiding to persist the preferences just because the key is null sounds a bit weak link. I would suggest adding a new attribute to this class, in the constructor, something like _persistPreferencesOnColumnChange_. Based on that you execute or not this: https://github.com/kiegroup/appformer/blob/master/uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java#L90

Reply:
@cristianonicolai absolutely agree, let me update this. Thanks